src/main/java/com/fortify/fod/Main.java

```diff
@@ -35,85 +35,126 @@ public static void main(String[] args) {
             System.exit(1);
         }
 
+        if (fc.bsiToken == null && (fc.url == null && ((fc.url.contains("http://") || fc.url.contains("https://"))))) {
+            System.err.println("Please provide Valid Fortify Url Parameter ( Example : https://ams.fotify.com/");
+            System.exit(1);
+        }
+
         if (!fc.hasApiCredentials() && !fc.hasUserCredentials()) {
-            System.out.println("The following options are required: -apiCredentials, -ac or -userCredentials, -uc");
+            System.err.println("The following options are required: -apiCredentials, -ac or -userCredentials, -uc");
             jc.usage();
             System.exit(1);
         }
 
-        if(fc.entitlementPreference == null) {
-            System.out.println("The entitlement preference option needs to be have the following values");
+        if (fc.hasUserCredentials() && fc.bsiToken == null && fc.tenantCode == null) {
```

Comment:
shouldn't it be `||`? you want to check that all are given.

Reply:
Nope , I don't need tenant code if bsi token is given and if user is using user credentials method for authentication , only if all conditions  are satisfied i need to display error message